molgenis-api-data/src/main/java/org/molgenis/api/data/v3/DataServiceV3Impl.java

```diff
@@ -72,6 +77,53 @@ public Entity find(String entityTypeId, String entityId, Selection filter, Selec
     return entity;
   }
 
+  @Transactional(readOnly = true)
+  @Override
+  public Entities findSubresources(
+      String entityTypeId,
+      String entityId,
+      String attributeName,
+      @Nullable @CheckForNull Query query,
+      Selection filter,
+      Selection expand,
+      Sort sort,
+      int size,
+      int number) {
+    Repository<Entity> repository = getRepository(entityTypeId);
+    EntityType entityType = repository.getEntityType();
+    Object typedEntityId = toTypedEntityId(entityType, entityId);
+
+    // Check if the attribute is suitable for this endpoint
+    Attribute refAttribute = entityType.getAttribute(attributeName);
+    if (refAttribute == null) {
+      throw new UnknownAttributeException(entityType, attributeName);
+    }
+    if (!EntityTypeUtils.isMultipleReferenceType(refAttribute)) {
+      throw new UnsupportedAttributeTypeException(refAttribute);
+    }
+
+    // get the mref ID's for this entity
+    EntityType refEntityType = refAttribute.getRefEntity();
+    Fetch refFetch = new Fetch().field(refEntityType.getIdAttribute().getName());
+    Fetch fetch = new Fetch().field(refAttribute.getName(), refFetch);
+    Entity entity = repository.findOneById(typedEntityId, fetch);
+    List<Object> refEntityIds =
+        StreamSupport.stream(entity.getEntities(attributeName).spliterator(), false)
+            .map(ref -> ref.getIdValue())
+            .collect(toList());
+
+    // Add 'in' query for the mref entity ID's
+    Repository<Entity> refRepository = getRepository(refEntityType.getId());
+    org.molgenis.data.Query<Entity> findQuery =
+        query != null ? queryMapperV3.map(query, refRepository) : new QueryImpl<>(refRepository);
+    if (!findQuery.getRules().isEmpty()) {
+      findQuery.and();
```

Comment:
- Please add a unit test for this case?
- What happens if the existing query is of form e.g. 'A or B' and you append 'and'? Nesting the existing query should solve possible problems.